docs/TuxReact.md

```diff
@@ -0,0 +1,449 @@
+`React` can be written beautiful [JSX](http://facebook.github.io/react/docs/displaying-data.html) which allows you to write React components as though they were HTML tags. We love JSX here at TuxedoJS and we stongly recommend using it. Our components are written using JSX but you do not need to write JSX with Tux. Our `React` examples in our docs will be written in JSX syntax.
+
+One exremely important note, if you want to use JSX with Tux (and you really should) you will need to call your `tux/React` variable `React`:
+
+```
+    var React = require('tux/React');
+```
+
+Why do we need to do this? Because JSX translates into `React.METHOD NAME` automatically, thus we need to call our variable `React` for our methods to work after compiling our JSX.
+
+In the following examples we will be writing in basic `React` syntax.
+
+***
+
+#### <a id="Rendering-in-React"></a>2) Rendering in React [#](#Rendering-in-React)
+The most common way to render a component in `React` is with the `React.render` method:
+
+```
+    var React = require('react');
+    var MyComponent = require('./components/MyComponent'); //Some component we want to render
+    React.render(<MyComponent />, document.body); //Using JSX for MyComponent
+```
+
+Oftentimes, if you are using the `tux/Router`, you will only need to invoke this method once to render the top level `Tux/Router/run` onto the DOM. This component will define your client-side application routes which will automatically render their child components when the corresponding route is reached:
+
+```
+    var React = require('tux/React');
+    var routes = require('./routes'); //your application routes
+    var run = require('tux/Router/run');
+
+    run(routes, function (Handler) {
+      React.render(<Handler />, document.body); //Again using JSX for Handler
+    });
+```
+
+At that point everything else will be rendered into the DOM through the `render` methods within your individual components. Let's dive into these `React` components.
+
+***
+
+#### <a id="React-Components"></a>3) React Components [#](#React-Components)
+`React` components are defined via the `React.createClass` method. Pass in an object to this method, and `React` will return a class which you can then use to render components built from that class into the DOM:
+
+```
+    var React = require('react');
+    var MyComponent = React.createClass({
+      render: function () {
+        return (
+          <h1>Hello World!</h1>
+        );
+      }
+    });
+```
+
+***
+
+#### <a id="React-Components-the-Render-Method"></a>3.1) the Render Method [#](#React-Components-the-Render-Method)
+All `React` classes must have a `render` method. The `render` method must return a single `React` component (either a basic `React` html component for rendering onto the DOM (`h1`, `div`, `p`, etc) or a `React` composite class (any custom build `React` component, for example `MyComponent` defined above is a `React` composite class).
```

Comment:
Recommend mentioning that the Render method must always return a valid HTML component, which means that it must be closed. eg \<input placeholder="things" />